pkg/plugins/manager/signature/manifest.go

```diff
@@ -326,16 +299,6 @@ func validateManifest(m PluginManifest, block *clearsign.Block) error {
 			return fmt.Errorf("%s is not a valid signature type", m.SignatureType)
 		}
 	}
-	keyring, err := openpgp.ReadArmoredKeyRing(bytes.NewBufferString(publicKeyText))
-	if err != nil {
-		return fmt.Errorf("%v: %w", "failed to parse public key", err)
-	}
 
-	if _, err = openpgp.CheckDetachedSignature(keyring,
-		bytes.NewBuffer(block.Bytes),
-		block.ArmoredSignature.Body, &packet.Config{}); err != nil {
-		return fmt.Errorf("%v: %w", "failed to check signature", err)
-	}
-
-	return nil
+	return manifestverifier.New(cfg, log.New("test")).Verify(m.KeyID, block)
```

Comment:
I might be wrong with the following. Given that `manifestverifier.New` is called for each validation the caching of keys in manifestverifier doesn't really make any difference I believe, it will lock/unlock for only one consumer and check an empty cache each time. Given it's behind a feature toggle should be fine for now.

Nit `log.New("test")` can be changed to something better. Maybe take a logger as argument to validateManifest and forward it to `manifestverifier.New`. 

Guess the manifestverifier might become a wire service eventually so that might clean things up so for now maybe just creating a logger with a more suitable name is the way forward?

Reply:
I have finally defined a service for the whole `signature` calculator. Please take a look!